### DISCUSSION TITLE
Displaying just the current space number?

### DISCUSSION THREAD

**Discussion by neutonfoo** [FelixKratz/SketchyBar#289] [Q&A] [answered]
Is it possible to have an item that just shows the current (mission control) space number? I'm not too interested in having an icon for each of my open spaces and looking for a solution to have one item which shows the current space index. (I figure I might be able to use the `space` type, but conditionally hide the other icons but seeing if there's a better alternative).

**Comment by FelixKratz** [comment, marked as answer]
Yeah, thats actually rather simple. You can use a regular item and subscribe it to the space change event:
You would need an item per display, this demo shows how it is done for the primary display:
```bash
sketchybar --add item space right \
           --set space script="$PLUGIN_DIR/current_space.sh" \
           --subscribe space space_change
```

`current_space.sh`:
```bash
#!/bin/sh

SPACE_ID=$(echo "$INFO" | jq -r '."display-1"')

sketchybar --set $NAME label=$SPACE_ID
```

You could go from here and choose to display the item only on the relevant display by assigning an `associated_display`.

**Reply by neutonfoo** [reply]
Ah wowow amazing! Thanks so much. I'm still reading through the docs and might have jumped the gun a little on this question! This is an amazing tool and can't wait to share my setup.
